src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLargeCombustionEngine.java

```diff
@@ -192,18 +192,19 @@ protected void formStructure(PatternMatchContext context) {
     }
 
     private boolean checkIntakesObstructed() {
-        EnumFacing facing = this.getFrontFacing();
-        boolean permuteXZ = facing.getAxis() == EnumFacing.Axis.Z;
-        BlockPos centerPos = this.getPos().offset(facing);
-        for (int x = -1; x < 2; x++) {
-            for (int y = -1; y < 2; y++) {
-                // Skip the controller block itself
-                if (x == 0 && y == 0)
+        for (int left = -1; left <= 1; left++) {
+            for (int up = -1; up <= 1; up++) {
+                if (left == 0 && up == 0) {
+                    // Skip the controller block itself
                     continue;
-                BlockPos blockPos = centerPos.add(permuteXZ ? x : 0, y, permuteXZ ? 0 : x);
-                IBlockState blockState = this.getWorld().getBlockState(blockPos);
-                if (!blockState.getBlock().isAir(blockState, this.getWorld(), blockPos))
+                }
+
+                final BlockPos checkPos = RelativeDirection.offsetPos(
+                        getPos(), getFrontFacing(), getUpwardsFacing(), isFlipped(), up, left, 1);
```

Comment:
Does flip matter here? Since the space being checked is essentially a square, like the rotor holder case

Reply:
It doesn't but it also doesn't hurt, figure since we have access to the setting easily, we may as well pass it in